bfvmm/src/vmcs/src/vmcs_intel_x64.cpp

```diff
@@ -33,8 +33,55 @@
 using namespace x64;
 using namespace intel_x64;
 
-vmcs_intel_x64::vmcs_intel_x64() :
-    m_vmcs_region_phys(0)
+void
+set_vm_control(uint32_t val, bool is_allowed0, bool is_allowed1,
```

Comment:
I would write this using template coding so that it's not so type specific, and so that it's definition can be in the .h. Also, it will likely change slightly with the comments above